pkg/auth/tokenparser_test.go

```diff
@@ -0,0 +1,111 @@
+package auth_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/auth"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTokenParser(t *testing.T) {
+	// create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+	configRegistry := configuration.CreateEmptyRegistry()
+
+	// create test keys
+	tokengenerator := testutils.NewTokenManager()
+	kid0 := uuid.NewV4().String()
+	_, err := tokengenerator.AddPrivateKey(kid0)
+	require.NoError(t, err)
+	kid1 := uuid.NewV4().String()
+	_, err = tokengenerator.AddPrivateKey(kid1)
+	require.NoError(t, err)
+
+	// startup public key service
+	keysEndpointURL := tokengenerator.NewKeyServer().URL
+
+	// set the config for testing mode, the handler may use this.
+	configRegistry.GetViperInstance().Set("testingmode", true)
+	assert.True(t, configRegistry.IsTestingMode(), "testing mode not set correctly to true")
+	// set the key service url in the config
+	configRegistry.GetViperInstance().Set("auth_client.public_keys_url", keysEndpointURL)
+	assert.Equal(t, keysEndpointURL, configRegistry.GetAuthClientPublicKeysURL(), "key url not set correctly")
+
+	// create KeyManager instance.
+	keyManager, err := auth.NewKeyManager(logger, configRegistry)
+	require.NoError(t, err)
+
+	// create TokenParser instance.
+	tokenParser, err := auth.NewTokenParser(logger, configRegistry, keyManager)
+	require.NoError(t, err)
+
+	t.Run("invalid arguments to new", func(t *testing.T) {
+		_, err := auth.NewTokenParser(nil, configRegistry, keyManager)
+		require.Error(t, err)
+		_, err = auth.NewTokenParser(logger, nil, keyManager)
+		require.Error(t, err)
+		_, err = auth.NewTokenParser(logger, configRegistry, nil)
+		require.Error(t, err)
```

Comment:
Check the error message.

@xcoulon I wonder if we can add a linter for that? :)

Reply:
fixed here: ca066720bf18117dcf3bdc5c8a6d7d884df5b233